docs/reference/sql/data-definition.md

```diff
@@ -0,0 +1,228 @@
+In practice, the limit is determined by the maximum message size that {{ site.ak }}
+can store and the resources dedicated to ksqlDB.
+
+## Streams
+
+A stream is a partitioned, immutable, append-only collection that represents a
+series of historical facts. For example, the rows of a stream could model a
+sequence of financial transactions, like "Alice sent $100 to Bob", followed by
+"Charlie sent $50 to Bob".
+
+Once a row is inserted into a stream, it can never change. New rows can be
+appended at the end of the stream, but existing rows can never be updated or
+deleted.
+
+Each row is stored in a particular partition. Every row, implicitly or explicitly,
+has a key that represents its identity. All rows with the same key reside in the
+same partition.
+
+To create a stream, use the `CREATE STREAM` command. The following example
+statement specifies a name for the new stream, the names of the columns, and
+the data type of each column.
+
+```sql
+CREATE STREAM s1 (
+    k VARCHAR KEY,
+    v1 INT,
+    v2 VARCHAR
+) WITH (
+    kafka_topic = 's1',
+    partitions = 3,
+    value_format = 'json'
+);
+```
+
+This creates a new stream named `s1` with three columns: `k`, `v1`, and `v2`.
+The column `k` is designated as the key of this stream, which controls the
+partition that each row is stored in. When the data is stored, the value
+portion of each row's underlying {{ site.ak }} record is serialized in the
+JSON format.
+
+Under the hood, each stream corresponds to a [Kafka topic](../../../overview/apache-kafka-primer/#topics)
+with a registered schema. If the backing topic for a stream doesn't exist when
+you declare it, ksqlDB creates it on your behalf, as shown in the previous
+example statement.
+
+You can also declare a stream on top of an existing topic. When you do that,
+ksqlDB simply registers its associated schema. If topic `s2` already exists,
+the following statement register a new stream over it:
+
+```sql
+CREATE STREAM s2 (
+    k1 VARCHAR KEY,
+    v1 VARCHAR
+) WITH (
+    kafka_topic = 's2',
+    value_format = 'json'
+);
+```
+
+!!! tip
+    When you create a stream on an existing topic, you don't need to declare
+    the number of partitions for the topic. ksqlDB infers the partition count
+    from the existing topic.
+
+## Tables
+
+A table is a mutable, partitioned collection that models change over time. In
+contrast with a stream, which represents a historical sequence of events, a
+table represents what is true as of "now". For example, you might use a table
+to model the locations where someone has lived as a stream: first Miami, then
+New York, then London, and so forth.
+
+Tables work by leveraging the keys of each row. If a sequence of rows shares a
+key, the last row for a given key represents the most up-to-date information
+for that key's identity. A background process periodically runs and deletes all
+but the newest rows for each key.
```

Comment:
I think we should be careful in our use of the terms `row` and `record` or `event`.

In my mind, `row` is referring to the row in the table, where are `record` or `event` is the record coming from the Kafka topic we're processing, that may mutate the row in the table.   

A non-temporal table only has one row with a specific key. Hence, IMHO, the phrase "If a sequence of rows share a key" is unintuitive ... as only one row can have a specific key.

```suggestion
Tables work by leveraging the keys of each row. If a sequence of records shares a
key, the last record for a given key represents the most up-to-date row
for that key's identity. A background process periodically runs and deletes all
but the newest record for each key.
```

Reply:
Maybe add link to compaction in the last sentence?